### PR TITLE
Fix mutex locking in ingest package and source.

### LINKED ISSUE #678: Seeing duplicates on packages
Just starting up playground and running `guacone files guac-data/docs/`. Seeing:

```json
{
  "node": {
    "__typename": "Package",
    "id": "1",
    "type": "oci",
    "namespaces": [
      {
        "id": "2",
        "namespace": "docker.io/library",
        "names": [
          {
            "id": "3",
            "name": "alpine",
            "versions": []
          }
        ]
      }
    ]
  }
}
```
```json
{
  "node": {
    "__typename": "Package",
    "id": "4",
    "type": "oci",
    "namespaces": [
      {
        "id": "5",
        "namespace": "docker.io/library",
        "names": [
          {
            "id": "6",
            "name": "alpine",
            "versions": [
              {
                "id": "7",
                "version": "sha256:1304f174557314a7ed9eddb4eab12fed12cb0cd9809e4c28f29af86979a3c870",
                "qualifiers": [
                  {
                    "key": "tag",
                    "value": "latest"
                  }
                ],
                "subpath": ""
              }
            ]
          }
        ]
      }
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by jeffmendoza**
Fixes #678

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]

File: pkg/assembler/backends/inmem/pkg.go [hunk 1]
```diff
@@ -246,92 +246,105 @@ func (p *pkgVersionNode) setPkgEquals(id uint32) { p.pkgEquals = append(p.pkgEqu
...
+	c.m.RLock()
 	namespacesStruct, hasNamespace := c.packages[input.Type]
+	c.m.RUnlock()
 	if !hasNamespace {
-		namespacesStruct = &pkgNamespaceStruct{
-			id:         c.getNextID(),
-			typeKey:    input.Type,
-			namespaces: pkgNamespaceMap{},
+		c.m.Lock()
+		namespacesStruct, hasNamespace = c.packages[input.Type]
```
Comment (pxp928) [inline pkg/assembler/backends/inmem/pkg.go:254]: is this duplicated code with line 250 above? Multiple instances of this...is this for a reason?

Comment (jeffmendoza) [reply pkg/assembler/backends/inmem/pkg.go:254]: The node could have been added between the release of the read mutex and aquiring the write mutex.

Comment (pxp928) [reply pkg/assembler/backends/inmem/pkg.go:254]: ah I see.

**Review by pxp928 — APPROVED** [review]

File: pkg/assembler/backends/inmem/pkg.go (hunk 1 above)
Comment (jeffmendoza) [reply pkg/assembler/backends/inmem/pkg.go:254]: https://github.com/guacsec/guac/pull/646#issuecomment-1485923075
It is messy to do the locks/unlocks and checking twice etc. that is why I did the re-entry on all the others.

**Pull request merged**